ui/src/pages/backoffice/Item/ItemForms/ItemEditor/components/ItemForm/ItemForm.js

```diff
@@ -24,12 +25,60 @@ import {
   serializeDocument,
   serializeInternalLocation,
 } from '@components/ESSelector/serializer';
+import { vocabulary as vocabularyApi } from '@api';
+import { withCancel } from '@api/utils';
 
 export class ItemForm extends Component {
   config = invenioConfig.items;
+  state = {
+    currencies: [],
+  };
+
+  componentDidMount() {
+    this.fetchCurrencies();
+  }
+
+  componentWillUnmount() {
+    this.cancellableFetchData && this.cancellableFetchData.cancel();
+  }
+
+  query = () => {
+    const searchQuery = vocabularyApi
+      .query()
+      .withType(invenioConfig.vocabularies.currencies)
+      .qs();
+    return vocabularyApi.list(searchQuery);
+  };
+
+  serializer = hit => ({
+    key: hit.metadata.key,
+    value: hit.metadata.key,
+    text: hit.metadata.key,
+  });
+
+  fetchCurrencies = async () => {
+    this.cancellableFetchData = withCancel(this.query());
+    try {
+      const response = await this.cancellableFetchData.promise;
+      const currencies = response.data.hits.map(hit => this.serializer(hit));
+      this.setState({ isLoading: false, currencies: currencies, error: null });
+    } catch (error) {
+      if (error !== 'UNMOUNTED') {
+        this.setState({
+          isloading: false,
+          options: [{ key: '', value: '', text: 'Failed to load currencies.' }],
```

Comment:
nit: I think I saw this multiple times maybe we can move it to invenioConfig

Reply:
https://github.com/inveniosoftware/invenio-app-ils/issues/871